pkg/http/roundtripper.go

```diff
@@ -0,0 +1,81 @@
+package http
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"unicode/utf8"
+)
+
+type bearerRoundTripper struct {
+	token   string
+	wrapper http.RoundTripper
+}
+
+func NewBearerRoundTripper(token string, rt http.RoundTripper) http.RoundTripper {
+	return &bearerRoundTripper{token: token, wrapper: rt}
+}
+
+func (rt *bearerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", rt.token))
+	return rt.wrapper.RoundTrip(req)
+}
+
+type debugRoundTripper struct {
+	next http.RoundTripper
+}
+
+func NewDebugRoundTripper(next http.RoundTripper) *debugRoundTripper {
+	return &debugRoundTripper{next}
+}
+
+func (rt *debugRoundTripper) RoundTrip(req *http.Request) (res *http.Response, err error) {
+	reqd, _ := httputil.DumpRequest(req, false)
+	reqBody := bodyToString(&req.Body)
+
+	res, err = rt.next.RoundTrip(req)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	resd, _ := httputil.DumpResponse(res, false)
+	resBody := bodyToString(&res.Body)
+
+	log.Printf("request url %v\n%s%s\n------ response\n"+
+		"%s%s\n======\n",
+		req.URL,
+		string(reqd), reqBody,
+		string(resd), resBody,
+	)
+
+	return
+}
+
+func bodyToString(body *io.ReadCloser) string {
+	if *body == nil {
+		return "<nil>"
+	}
+
+	var b bytes.Buffer
+	_, err := b.ReadFrom(*body)
+	if err != nil {
+		panic(err)
```

Comment:
I don't think there is any reason to panic here. Nothing is forcing the `bodyToString` signature to be `string` rather than `(string, error)`. Especially given that this is _only_ used in the `debugRoundTripper` type, we should return the errors and log them in the calling func.